src/main/java/com/capitalone/dashboard/service/CollectorItemServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+        }
+
+        Optional<Collector> collector = collectorRepository.findByCollectorTypeAndName(collectorType, collectorName).stream().findFirst();
+        if (!collector.isPresent()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Error :: Could not find collector: " + collectorName);
+        }
+
+        // get collectorItems that have not been updated since the endDate given
+        List<CollectorItem> collectorItems = collectorItemRepository.findByCollectorIdAndLastUpdatedBefore(collector.get().getId(), endDate);
+        LOG.info(String.format("deleteDisconnectedItems :: Found %d collectorItems to verify", collectorItems.size()));
+
+        // iterate through enabled items and check if they are connected to a dashboard
+        for (CollectorItem collectorItem : collectorItems) {
+            if (!hasComponent(collectorType, collectorItem.getId())) {
+
+                String loggingPrefix = String.format("findOldCollectorItems :: Removing (#%d of %d):: could not find a dashboard for the collectorItem with the following options:"
+                        ,collectorItems.indexOf(collectorItem)+1, collectorItems.size());
+
+                logDeletedCollectorItem(collectorType, collectorItem, loggingPrefix);
+                collectorItemRepository.delete(collectorItem.getId());
+                count++;
+            }
+        }
+
+        LOG.info(String.format("deleteDisconnectedItems :: Finished (duration=%s) :: collectorType=%s :: Found %d items with no corresponding dashboard.", System.currentTimeMillis() - startTime, collectorType.toString(), count));
+        return ResponseEntity.status(HttpStatus.OK).body(String.format("Successfully removed %d  %s type collectorItems out of %d", count, collectorType.toString(), collectorItems.size()));
+    }
+
+
+    /**
+     * ******************************************************************************
+     * Bottom three methods are helper functions for the deleteDisconnectedItems()
+     * ******************************************************************************
+     */
+    private Boolean isCleanableCollector(CollectorType collectorType){
+        return (collectorType == LibraryPolicy || collectorType == StaticSecurityScan ||
+                collectorType == SCM);
+    }
+
+
+    private Boolean hasComponent(CollectorType collectorType, ObjectId collectorItemId){
+        switch (collectorType){
+            case LibraryPolicy:
+                return componentRepository.findByLibraryPolicyCollectorItems(collectorItemId).stream().findFirst().isPresent();
+            case SCM:
+                return componentRepository.findBySCMCollectorItemId(collectorItemId).stream().findFirst().isPresent();
+            case StaticSecurityScan:
+                return componentRepository.findByStaticSecurityScanCollectorItems(collectorItemId).stream().findFirst().isPresent();
+            default:
+                return true;
```

Comment:
default value is contradictory with method name. plz update either methodName or default value.